File: Evl_fullstack-front/src/app/restaurant/restaurant-dto.ts
export interface RestaurantDto {
  id: number;
  nom: string;
  adresse: string;
  moyenne: number;
  tags?: TagDto[];
  evaluations?: EvaluationDto[];
  evaluationFinale?: EvaluationFinaleDto;
  illustration?: string;
}

export interface TagDto {
    id: number;
    nom: string;
}

export interface EvaluationDto {
    id: number;
    nom: string;
    note: number;
    commentaire: string;
    dateCreation: Date;
    dateModification: Date;
    photos?: string[];
}

export interface EvaluationFinaleDto {
    id: number;
    nom: string;
    note: number;
    description: string;
}

export interface AddRestaurantDto {
    nom: string;
    adresse: string;
    tags: TagDto[];
}

File: Eval_fullstack/src/main/java/fr/Eval_fullstack/dto/response/RestaurantDto.java
package fr.Eval_fullstack.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RestaurantDto {

    @JsonProperty("id")
    private Integer id;

    @JsonProperty("nom")
    private String nom;

    @JsonProperty("adresse")
    private String adresse;

    @JsonProperty("moyenne")
    private float moyenne;

    @JsonProperty("evaluations")
    private List<EvaluationDto> evaluations;

    @JsonProperty("evaluationFinale")
    private EvaluationFinaleDto evaluationFinale;

    @JsonProperty("illustrations")
    private List<String> illustrations;

    @JsonProperty("tags")
    private List<String> tags;

}
